src/components/application_manager/src/message_helper.cc

```diff
@@ -2394,41 +2394,39 @@ mobile_apis::Result::eType MessageHelper::ProcessSoftButtons(
       return Result::DISALLOWED;
     }
 
+    bool is_image_valid = false;
+    bool is_image_exists = request_soft_buttons[i].keyExists(strings::image);
+
+    bool is_text_valid = false;
+    bool is_text_exists = request_soft_buttons[i].keyExists(strings::text);
+
+    if (is_image_exists) {
+      is_image_valid =
+          Result::SUCCESS ==
+          VerifyImage(request_soft_buttons[i][strings::image], app, app_mngr);
+    }
+
+    if (is_text_exists) {
+      is_text_valid = !request_soft_buttons[i][strings::text].empty() &&
```

Comment:
@indlu Create separate variable for `request_soft_buttons[i][strings::text]`inside if.